### ISSUE TITLE
[google_maps_flutter] Setting custom semantic label for each map marker

### ISSUE THREAD

**Issue by suesitran** [flutter/flutter#114895]
## Use case
I'm implementing an app what uses Google map, and I want to provide accessibility support for all markers in the map. Currently the SR reads map marker as 'button' on iOS, and there's no way to set semantic label to map marker.

I want to be able to set different semantic label for each marker in the map.

## Proposal
Create a field for semantic label in Marker class, and if this field is set, then SR will announce this marker as '<label>, button' for iOS, or '<label>, double tap to activate' for Android

**Comment by mastermakrela** [comment]
`GMSMarker` supports this in iOS SDK through [`title`](https://developers.google.com/maps/documentation/ios-sdk/reference/interface_g_m_s_overlay#a04f4c8cc1cbf24ff6a354820049ca604) property of `GMSOverlay`.

But in flutter wrapper, this property seams to be set by `InfoWindow`s `title`. 

So you can set pins semantic label like this:

```dart
Marker(
  markerId: MarkerId('marker_1'),
  infoWindow: InfoWindow(title: "I'm a pin"),
)
```

This will be read as: _I'm a pin - button_

The only side effect of this is that the Info Window will appear on tap, but it might be better than no accessibility at all.

**Comment by cedvdb** [comment]
infoWindows doesn't read in talkback in android
